@xen-orchestra/lite/src/components/ActionButton.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import LoadingButton, { LoadingButtonProps } from '@mui/lab/LoadingButton'
+import { withState } from 'reaclette'
+
+interface ParentState {}
+
+interface State {
+  isLoading: boolean
+}
+
+interface Props extends LoadingButtonProps {
+  onClick: (e: React.MouseEvent) => Promise<void>
+}
+
+interface ParentEffects {}
+
+interface Effects {
+  _onClick: React.MouseEventHandler<HTMLButtonElement>
+}
+
+interface Computed {}
+
+const ActionButton = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({ isLoading: false }),
+    effects: {
+      _onClick: function (e) {
+        this.state.isLoading = true
+        return this.props.onClick(e).finally(() => (this.state.isLoading = false))
+      },
+    },
+  },
+  ({ children, color = 'secondary', effects, onClick, resetState, state, variant = 'contained', ...props }) => (
+    <LoadingButton
+      color={color}
+      disabled={state.isLoading}
+      fullWidth
```

Comment:
- Why? 
- We should have the possibility to change its size, no?

Reply:
`FullWidth` mean, the button will take all the place available in the parent. We still can change this size by change the contenaire size.
It's exactly the same behaviour that the other buttons.